packages/svelte/src/internal/client/runtime.js

```diff
@@ -1012,7 +1012,13 @@ function get_parent_context(component_context) {
  */
 export function update(signal, d = 1) {
 	const value = get(signal);
-	set(signal, value + d);
+	let to_mutate = value;
+	if (d === 1) {
+		to_mutate++;
+	} else {
+		to_mutate--;
+	}
+	set(signal, to_mutate);
```

Comment:
this can be simplified a lot, we just need to coerce the original value

Reply:
Help